nested_K8sCustomIngressTlsHostsHaveDefinedDomainSuffix.bicep

```diff
@@ -0,0 +1,159 @@
+targetScope = 'subscription'
+
+/*** PARAMETERS ***/
+
+@description('The identifying name this custom policy is deployed with.')
+@minLength(1)
+param policyName string = 'K8sCustomIngressTlsHostsHaveDefinedDomainSuffix' 
```

Comment:
Why make this a param with a default?  You don't use the default, and you provide this exact value, and the file is named is this, and the base64 is encoded text basically uses this as well.  I'd just simplify and hard code it on line 13 so this isn't a one-off.

Reply:
My first attempt was exactly as you suggested: Use the policyId from the nested output as first parameter to the `guid(.)` function in `cluster-stamp.bicep` (this would even be the most consistent setup copared to the other policy assignments that also use the policy ID here).

However, the Bicep compiler rejected the usage of `modK8sIngressTlsHostsHaveDefinedDomainSuffix.outputs` in `name: guid(...)` when calling `az deployment group create -g rg-bu0001a0008 -f cluster-stamp.bicep...` the with following message:

```
Error BCP120: This expression is being used in an assignment to the "name" property of the "Microsoft.Authorization/policyAssignments" type, which requires a value that can be calculated at the start of the deployment. Properties of modK8sIngressTlsHostsHaveDefinedDomainSuffix which can be calculated at the start include "name".
```

So I was looking for another value to the `guid(.)` function which would be a sufficiently unique to this assignment to avoid guid collisions with other resources. The best idea I came up with was to pull the name (that should be fairly unique) out as parameter, define it in `cluster-stamp.bicep` and use as parameter to the module (to avoid divergence of the name in policyAssignment and policyDefinition). (Although I agree that it technically makes no sense to declare a parameter here.)

Just to be sure I did not miss anything, I checked out your suggestion and tried to deploy again but got the same message.

Let me know what you think...

Reply:
Bummer, I was wondering if that was going to happen.  Well, the hard coded policy name in there as the first param should be good enough, That's unfortunate, but not really the end of the world :)

Reply:
Okay. Sounds good.